src/librustdoc/formats/cache.rs

```diff
@@ -95,7 +95,9 @@ pub(crate) struct Cache {
     // Private fields only used when initially crawling a crate to build a cache
     stack: Vec<Symbol>,
     parent_stack: Vec<DefId>,
+    impl_generics_stack: Vec<(clean::Type, clean::Generics)>,
```

Comment:
I'd really like to get rid of all these stacks at some point. Is it possible to avoid adding this one?